ground/src/main/java/com/google/android/ground/ui/map/gms/mog/UrlInputStream.kt

```diff
@@ -38,19 +39,25 @@ class UrlInputStream(private val url: String, private val byteRange: LongRange?
 
   private fun openStream(): InputStream {
     val urlConnection = URL(url).openConnection() as HttpURLConnection
-    urlConnection.requestMethod = "GET"
-    urlConnection.readTimeout = READ_TIMEOUT_MS
-    if (byteRange != null) {
-      urlConnection.setRequestProperty("Range", "bytes=${byteRange.first}-${byteRange.last}")
+    try {
+      urlConnection.requestMethod = "GET"
+      urlConnection.connectTimeout = CONNECT_TIMEOUT_MS
+      urlConnection.readTimeout = READ_TIMEOUT_MS
+      if (byteRange != null) {
+        urlConnection.setRequestProperty("Range", "bytes=${byteRange.first}-${byteRange.last}")
+      }
+      urlConnection.connect()
+      val responseCode = urlConnection.responseCode
+      if (responseCode == 404) throw FileNotFoundException("$url not found")
+      val expectedResponseCode = if (byteRange == null) 200 else 206
+      if (responseCode != expectedResponseCode) {
+        throw IOException("HTTP $responseCode accessing $url")
+      }
+      return urlConnection.inputStream
+    } catch (e: Exception) {
+      urlConnection.disconnect()
+      error("Attempt Failed with $e")
```

Comment:
Won't this throw another exception? How are we planning to handle the timeouts?

Reply:
1. As for this pr, the idea is to get logs
2. Added connectTimeout
3. for timeout, or any other failure, it should wrap the failure and return - a part of https://github.com/google/ground-android/issues/2891

Reply:
> Why do we need urlConnection.disconnect()? What happens if that call fails?

For releasing the resources and not waiting for the garbage collection to clean it, hence doing it ourselves.
https://developer.android.com/reference/java/net/HttpURLConnection#disconnect()

Reply:
To your question(s):

* The `catch` block catches the exceptions explicitly thrown in the `try` block, as well as anything else thrown by functions called inside the `try` block. 
* Calling `error()` swallows throws an `IllegalStateException`, in this case with the original error in stack trace packed inside that exceptions `message`.
* If the intent of the `catch` block is to close the connection on error, and that it will be up to the caller to log exceptions, you should just `throw e` instead to allow that to happen.
* If instead the intention is for this method to handle all exceptions, then we should log the error with `Timber.d` and return a failure state.

More generally, we should be consistent about who is responsible for closing the connection. Since the caller will need to close the connection once they've consumed the stream, then closing the connection should be done there.